src/execution/index/art/art.cpp

```diff
@@ -204,6 +204,160 @@ void ART::GenerateKeys(DataChunk &input, vector<unique_ptr<Key>> &keys) {
 	}
 }
 
+struct KeySection {
+	KeySection(idx_t start_p, idx_t end_p, idx_t depth_p, data_t key_byte_p)
+	    : start(start_p), end(end_p), depth(depth_p), key_byte(key_byte_p) {};
+	idx_t start;
+	idx_t end;
+	idx_t depth;
+	data_t key_byte;
+};
+
+void GetNodeChildren(vector<KeySection> &child_sections, vector<unique_ptr<Key>> &keys, KeySection &key_section) {
```

Comment:
Should I maybe rename this to `GetChildSections`? I am not yet completely happy with the naming, any suggestions? Also, is `KeySection` a good choice?